packages/astro/src/core/preview/index.ts

```diff
@@ -23,100 +22,73 @@ export interface PreviewServer {
 	stop(): Promise<void>;
 }
 
-type SendStreamWithPath = send.SendStream & { path: string };
-
-function removeBase(base: string, pathname: string) {
-	if (base === pathname) {
-		return '/';
-	}
-	let requrl = pathname.substr(base.length);
-	return prependForwardSlash(requrl);
-}
-
 /** The primary dev action */
 export default async function preview(config: AstroConfig, { logging }: PreviewOptions): Promise<PreviewServer> {
+	// polyfill WebAPIs to globalThis for Node v12, Node v14, and Node v16
+	polyfill(globalThis, {
```

Comment:
Why is this needed?

Reply:
It was needed when I was using `.at(-1)`. I found another way to do it and removed the code.